lexicon/providers/gandi.py

```diff
@@ -0,0 +1,277 @@
+
+    The class will use the following environment variables to configure
+    it instance. For more information, read the Lexicon documentation.
+
+    - LEXICON_GANDI_API_ENDPOINT - the Gandi API endpoint to use
+      The default is the production URL https://rpc.gandi.net/xmlrpc/.
+      Set this environment variable to the OT&E URL for testing.
+
+    """
+
+    def __init__(self, options, provider_options=None):
+        """Initialize Gandi DNS provider."""
+
+        super(Provider, self).__init__(options)
+
+        if provider_options is None:
+            provider_options = {}
+
+        api_endpoint = provider_options.get('api_endpoint') or 'https://rpc.gandi.net/xmlrpc/'
+
+        self.apikey = self.options['auth_token']
+        self.api = xmlrpclib.ServerProxy(api_endpoint, allow_none=True)
+
+        # self.domain_id is required by test suite
+        self.domain_id = None
+        self.zone_id = None
+
+        self.domain = self.options['domain'].lower()
+
+    # Authenicate against provider,
+    # Make any requests required to get the domain's id for this provider,
+    # so it can be used in subsequent calls. Should throw an error if
+    # authentication fails for any reason, or if the domain does not exist.
+    def authenticate(self):
+        """Determine the current domain and zone IDs for the domain."""
+
+        try:
+            payload = self.api.domain.info(self.apikey, self.domain)
+            self.domain_id = payload['id']
+            self.zone_id = payload['zone_id']
+
+        except xmlrpclib.Fault as err:
+            raise Exception("Failed to authenticate: '{0}'".format(err))
+
+    # Create record. If record already exists with the same content, do nothing'
+    def create_record(self, type, name, content):
+        """Creates a record for the domain in a new Gandi zone."""
+
+        version = None
+        ret = False
+
+        name = self._canonicalize_name(name)
+
+        # This isn't quite "do nothing" if the record already exists.
+        # In this case, no new record will be created, but a new zone version
+        # will be created and set.
+        try:
+            version = self.api.domain.zone.version.new(self.apikey, self.zone_id)
+            self.api.domain.zone.record.add(self.apikey, self.zone_id, version,
+                                            {'type': type.upper(),
+                                             'name': name,
+                                             'value': content,
+                                             'ttl': self.options.get('ttl') or self.default_ttl
+                                            })
+            self.api.domain.zone.version.set(self.apikey, self.zone_id, version)
+            ret = True
+
+        finally:
+            if not ret and version is not None:
+                self.api.domain.zone.version.delete(self.apikey, self.zone_id, version)
+
+        print("create_record: {0}".format(ret))
```

Comment:
can you replace all print statements with `logger.debug` usage? https://github.com/AnalogJ/lexicon/blob/master/lexicon/providers/cloudflare.py#L75 is a good example of how that's done in other providers.